server_addon/houdini/client/ayon_houdini/api/pipeline.py

```diff
@@ -85,10 +85,9 @@ def install(self):
             # initialization during start up delays Houdini UI by minutes
             # making it extremely slow to launch.
             hdefereval.executeDeferred(shelves.generate_shelves)
-
-        if not IS_HEADLESS:
-            import hdefereval # noqa, hdefereval is only available in ui mode
             hdefereval.executeDeferred(creator_node_shelves.install)
+            if os.environ.get("AYON_WORKFILE_TOOL_ON_START"):
+                hdefereval.executeDeferred(lib.wait_startup_launch_workfiles_app)
```

Comment:
This works fine for me without all the wait logic - making this PR much simpler.
```suggestion
            if os.environ.get("AYON_WORKFILE_TOOL_ON_START"):
                def show_workfiles_tool():
                    # Make sure on top is enabled on first show so the
                    # window is not hidden under main nuke window
                    from ayon_core.tools.utils import host_tools
                    host_tools.show_workfiles(parent=hou.qt.mainWindow(),
                                              on_top=True)

                hdefereval.executeDeferred(show_workfiles_tool)

```

Reply:
I used this if you don't mind
```python
if os.environ.get("AYON_WORKFILE_TOOL_ON_START"):
    hdefereval.executeDeferred(lambda: host_tools.show_workfiles(parent=hou.qt.mainWindow()))
```